test/extensions/filters/http/buffer/buffer_filter_integration_test.cc

```diff
@@ -63,6 +63,35 @@ TEST_P(BufferIntegrationTest, RouterRequestPopulateContentLength) {
   EXPECT_EQ("200", response->headers().Status()->value().getStringView());
 }
 
+TEST_P(BufferIntegrationTest, RouterRequestPopulateContentLengthOnTrailers) {
+  config_helper_.addFilter(ConfigHelper::DEFAULT_BUFFER_FILTER);
+  initialize();
+
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+  auto encoder_decoder = codec_client_->startRequest(Http::TestHeaderMapImpl{
+      {":method", "POST"}, {":scheme", "http"}, {":path", "/shelf"}, {":authority", "host"}});
+  request_encoder_ = &encoder_decoder.first;
+  IntegrationStreamDecoderPtr response = std::move(encoder_decoder.second);
+  codec_client_->sendData(*request_encoder_, "0123", false);
+  codec_client_->sendData(*request_encoder_, "456", false);
+  codec_client_->sendData(*request_encoder_, "789", false);
+  Http::TestHeaderMapImpl request_trailers{{"request", "trailer"}};
+  codec_client_->sendTrailers(*request_encoder_, request_trailers);
```

Comment:
@alyssawilk with and empty map I got an asan failure because of taking a reference from a nullptr here:
https://github.com/envoyproxy/envoy/blob/cc8fe6d34d7fe04339a6c422a40f429d96008389/source/common/http/http2/codec_impl.cc#L216

Things obviously end up fine because nghttp2 first checks the array length. Do you think it is worth doing something envoy side so that we can send empty trailers without tripping the address sanitizer?

Reply:
I'm also confused how content length changes here result in empty trailers.  Can you clue me in a bit more?

Reply:
oh sorry, I meant that when I originally wrote the test by sending an empty trailers map (https://github.com/envoyproxy/envoy/pull/8609/commits/78f6b39b6b23cdb522c3da81d632956069e3b0a8#diff-13f71f37c14f0986b566cbf97601bf44R78) I was getting an ASAN failure. So not an effect of the patch, but rather of how I wrote the test.

Reply:
Oh gotcha!  Yeah, if sending an empty trailers map causes ASAN bugs, we should file and fix that as a separate PR lest some filters clear the only trailers entry and cause problems
@mattklein123 for visibility.

Reply:
reference issue: https://github.com/envoyproxy/envoy/issues/8720.